src/main/java/mate/academy/book/BookDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+        int variablePlace = 1;
+        String sqlGetStatement = "SELECT * FROM book WHERE id = ?";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sqlGetStatement)) {
+            statement.setLong(variablePlace,id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Long idFromDB = resultSet.getObject("id", Long.class);
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+                Book book = new Book(idFromDB,title,price);
+                return Optional.of(book);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant get book with id:" + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sqlFindAllStatement = "SELECT * FROM book";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sqlFindAllStatement);
+                ResultSet resultSet = statement.executeQuery()) {
+
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+
+                Book book = new Book(id, title, price);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't read data", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        int idPlace = 3;
+        int titlePlace = 1;
+        int pricePlace = 2;
+        String sqlUpdateStatement = "UPDATE book SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sqlUpdateStatement)) {
+            statement.setLong(idPlace,book.getId());
+            statement.setString(titlePlace,book.getTitle());
+            statement.setBigDecimal(pricePlace,book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows > 0) {
+                return book;
+            } else {
+                throw new RuntimeException("Failed to update book with id: " + book.getId());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update " + book.toString(), e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        int idPlace = 1;
+        String sqlDeleteStatement = "DELETE FROM book WHERE id = ? ";
```

Comment:
In the `deleteById` method, ensure that the SQL statement uses the correct table name `books` as specified in the task description. The current statement uses `book`, which may lead to errors if the table name is incorrect.